tests/unit/s2n_malformed_handshake_test.c

```diff
@@ -483,7 +489,7 @@ int main(int argc, char **argv)
     EXPECT_EQUAL(memcmp(conn->pending.server_random, zero_to_thirty_one, 32), 0);
 
     /* Check that the server hello message was not processed, we're stuck in SERVER_CERT */
-    EXPECT_EQUAL(conn->handshake.state, SERVER_CERT);
+    EXPECT_EQUAL(conn->handshake.message_number, 2);
```

Comment:
I think we can use the "ACTIVE_STATE" macro here to assert the exact state.
`EXPECT_EQUAL(ACTIVE_STATE(conn), SERVER_CERT)`